lib/nanoc/base/entities/configuration.rb

```diff
@@ -47,6 +51,21 @@ def with_defaults
       self.class.new(new_wrapped)
     end
 
+    def with_environment(env = nil)
+      return self unless @wrapped.key? :environments
+
+      # Set active environment
+      env ||= ENV.fetch 'NANOC_ENVIRONMENT', :default
```

Comment:
1. Same parentheses remark as above.

2. I suggest `NANOC_ENV`, for consistency with `RAILS_ENV` and `RACK_ENV`